core/sysparam.c

```diff
@@ -719,97 +761,65 @@ sysparam_status_t sysparam_get_string(const char *key, char **destptr) {
     return SYSPARAM_OK;
 }
 
-sysparam_status_t sysparam_get_int(const char *key, int32_t *result) {
-    char *buffer;
-    char *endptr;
+sysparam_status_t sysparam_get_int32(const char *key, int32_t *result) {
     int32_t value;
+    size_t actual_length;
+    bool is_binary;
     sysparam_status_t status;
 
-    status = sysparam_get_string(key, &buffer);
+    status = sysparam_get_data_static(key, (uint8_t *)&value, sizeof(int32_t),
+                                      &actual_length, &is_binary);
     if (status != SYSPARAM_OK) return status;
-    value = strtol(buffer, &endptr, 0);
-    if (*endptr) {
-        // There was extra crap at the end of the string.
-        free(buffer);
+    if (!is_binary || actual_length != sizeof(int32_t))
         return SYSPARAM_PARSEFAILED;
-    }
-
     *result = value;
-    free(buffer);
-    return SYSPARAM_OK;
+    return status;
 }
 
-sysparam_status_t sysparam_get_bool(const char *key, bool *result) {
-    char *buffer;
+sysparam_status_t sysparam_get_int8(const char *key, int8_t *result) {
```

Comment:
I would like to add that the inclusion of sysparam_get_bool sort of was agreed upon when the original PR got merged. This does not mean it is written in stone and must remain that way forever of course. As I commented on one of your PRs, it is a good practice to keep PRs short and, how should I put it, singleton. You have some very good changes in this PR that improves the sysparam framework but those got blocked from being merged due to other parts needing discussion. Had the PR been broken up, we could have merged the good parts without further discussion.